### PR TITLE
Fix hardcoded link versions: v23.1 through v24.2

### PR DESCRIPTION
**Pull request by rmloveland**
A number of docs pages still have hardcoded links in the form `{% link v23.2/... %}`

This PR fixes them to point at the version defined in `{{page.version.version}}`

Additionally, we fix some links to includes that are sourced from both Cloud and on-prem docs that were hardcoded. In order to fix these, we had to add some Liquid logic to check the page titles.

### CODE REVIEW

File: src/current/cockroachcloud/backup-and-restore-overview.md [hunk 1]
```diff
@@ -217,4 +217,4 @@ For practical examples of running backup and restore jobs, watch the following v
 
 - Considerations for using [backup](https://www.cockroachlabs.com/docs/{{site.current_cloud_version}}/backup#considerations) and [restore](https://www.cockroachlabs.com/docs/{{site.current_cloud_version}}/restore#considerations).
 - [Backup collections](https://www.cockroachlabs.com/docs/{{site.current_cloud_version}}/take-full-and-incremental-backups#backup-collections) for details on how CockroachDB stores backups.
-- [Restoring backups](https://www.cockroachlabs.com/docs/{{site.current_cloud_version}}/restoring-backups-across-versions) across major versions of CockroachDB.
\ No newline at end of file
+- [Restoring backups](https://www.cockroachlabs.com/docs/{{site.current_cloud_version}}/restoring-backups-across-versions) across major versions of CockroachDB.
```
Comment (kathancox) [inline src/current/cockroachcloud/backup-and-restore-overview.md:220]: Not sure what changed here?

**Review by kathancox — APPROVED** [review]
From what I can tell this looks good! Thanks!

File: src/current/cockroachcloud/backup-and-restore-overview.md (hunk 1 above)
Comment (rmloveland) [reply src/current/cockroachcloud/backup-and-restore-overview.md:220]: my editor added a newline at the end of the file

from what I can tell some other editors remove newline at end of file

at some point we could probably consider setting up a team-wide [editorconfig](https://editorconfig.org) for these sorts of things

I didn't find an issue for that so I filed this: https://cockroachlabs.atlassian.net/browse/DOC-10825

**Comment by rmloveland** [comment]
> From what I can tell this looks good! Thanks!

yayyy thanks @kathancox !

File: src/current/cockroachcloud/backup-and-restore-overview.md (hunk 1 above)
Comment (kathancox) [reply src/current/cockroachcloud/backup-and-restore-overview.md:220]: Ah cool! Sorry missed the new line! Thanks for doing this!

**Pull request merged**